src/switchroot/ostree-prepare-root.c

```diff
@@ -44,6 +44,25 @@
 
 #include "ostree-mount-util.h"
 
+static void
+die (void)
+{
+  if (getpid() == 1)
+    {
+      /* We're init - Give the user a chance to recover */
+      fprintf (stderr, "init: Entering recovery shell\n");
```

Comment:
I could imagine rather than offering an interactive shell, we do an "info dump" of things like the mount points and existing files.  I've always wished the kernel would show me `lsblk -o NAME,UUID` when failing to mount the rootfs for example - and I suspect that type of thing happens more often than this bit of ostree will break.

So it feels to me like the simplest thing for now is to rebase and drop this patch out - if you want we can keep it as a separate PR?